react/features/desktop-picker/components/DesktopPicker.js

```diff
@@ -112,20 +106,19 @@ class DesktopPicker extends Component {
         this._onPreviewClick = this._onPreviewClick.bind(this);
         this._onSubmit = this._onSubmit.bind(this);
         this._updateSources = this._updateSources.bind(this);
+
+        const { desktopSharingSources } = this.props.options;
+
+        this.state.types = this._getValidTypes(desktopSharingSources);
     }
 
     /**
-     * Perform an immediate update request for DesktopCapturerSources and begin
-     * requesting updates at an interval.
+     * Starts polling.
      *
      * @inheritdoc
+     * @returns {void}
      */
-    componentWillMount() {
```

Comment:
You started polling after mount to ensure the component has been rendered once for additional re-renders?

Reply:
I read the React documentation that it is not recommended to do side effects in `componentWillMount`, they are recommending `componentDidMount`.